kobo-uncaged/main.go

```diff
@@ -497,10 +497,16 @@ func (ku *KoboUncaged) saveCoverImage(contentID string, size image.Point, imgB64
 
 	for _, cover := range []koboCover{fullCover, libFull, libGrid} {
 		nsz := cover.Resize(ku.device, sz)
-		nimg := imaging.Resize(img, nsz.X, nsz.Y, imaging.Linear)
 		nfn := filepath.Join(imgDir, cover.RelPath(imgID))
 
-		log.Printf("Resized %s cover to %s (target %s) for %s\n", sz, nsz, cover.Size(ku.device), cover)
+		log.Printf("Resizing %s cover to %s (target %s) for %s\n", sz, nsz, cover.Size(ku.device), cover)
+
+		nimg := img
+		if !sz.Eq(nsz) {
+			nimg = imaging.Resize(nimg, nsz.X, nsz.Y, imaging.Linear)
+		} else {
+			log.Printf(" -- Skipped resize: already correct size")
```

Comment:
-> `log.Println`?

Reply:
Otherwise, good call on making it generic, that was on my notes for today ;).

Reply:
I personally tend to prefer using printf for everything, as I find it easier to read (figuring out where I put newlines and where I didn't). But, for consistency, I'll change it here.

Reply:
I agree with you, but the lack of a trailing LF confused me ;).